### PR TITLE
Fix release instructions

### PR DESCRIPTION
**Pull request by darrnshn**
This patch updates the release instructions in `CONTRIBUTING.md`:
* Add instructions to get fresh clone of repo.
* Added `--ignore-unmatch` to `git rm .gitignore` because `.gitignore` may not exist.
* Clarify where to push things.
* Replaced `git push HEAD:refs/heads/master` with `git push origin master` because the former simply did not work.
* Add more detailed instructions for drafting the release on GitHub.

Note that this diff contains an irrelevant commit `Fix missing web-animations.min.js issue`. This will disappear once that commit is pushed to `dev` by @ewilligers .

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,14 +85,17 @@ Example: `http://localhost:9876/base/test/web-platform-tests/web-animations/anim
     grunt
     # Optional "grunt test" to make sure everything still passes.
     git add -f *.min.js{,.map}
-    git rm .gitignore
+    git rm --ignore-unmatch .gitignore
     git commit -m 'Add build artifacts from '`cat .git/refs/heads/dev`
-    git push HEAD:refs/heads/master
+    git push origin master
```
Comment (alancutter) [inline CONTRIBUTING.md]: Let's add a step to verify that `git remote -v` shows that origin points to `github.com/web-animations/web-animations-js`.

File: History.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+### 2.3.1 - *July 20 2017*
+
+  * Fix [https://github.com/web-animations/web-animations-js/issues/157](missing web-animations.min.js issue)
```
Comment (alancutter) [inline History.md]: This change is unrelated and should be kept separate.

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
     "type": "git",
     "url": "https://github.com/web-animations/web-animations-js.git"
   },
-  "version": "2.3.0",
+  "version": "2.3.1",
```
Comment (alancutter) [inline package.json]: This change is unrelated and should be kept separate.

File: CONTRIBUTING.md (hunk 1 above)
Comment (alancutter) [inline CONTRIBUTING.md]: Steps "Add versioned release notes to `History.md`" and "Commit the above changes" should probably also be using origin.

File: History.md (hunk 2 above)
Comment (darrnshn) [inline History.md]: Yep, it looks like I didn't actually push to origin dev (not sure if I was meant to).
@ewilligers will be pushing that commit to origin dev soon so this diff should disappear.

File: CONTRIBUTING.md (hunk 1 above)
Comment (darrnshn) [inline CONTRIBUTING.md]: Hmm what happens if their `git remote -v` doesn't show the right value? If it's a fork, I'm guessing it'll point to their own forked repo?

Comment (alancutter) [inline CONTRIBUTING.md]: In that case the least invasive way to resolve it is to make a new clone of the central repo and work in that.

File: History.md (hunk 2 above)
Comment (alancutter) [inline History.md]: Mmm, the instructions should probably say to push to origin/dev to avoid this problem in future.

File: CONTRIBUTING.md (hunk 1 above)
Comment (darrnshn) [inline CONTRIBUTING.md]: Cool, so maybe for releases, we use a clone, but for code changes, use fork? I'll add a step in the beginning for cloning the repo.

Comment (alancutter) [inline CONTRIBUTING.md]: Sure that works. If the release process doesn't assume the existence of a repository that gives it more guarantees on the repo's state.

Comment (darrnshn) [inline CONTRIBUTING.md]: Cool, uploaded new commits, PTAL
